gatling-jms/src/main/scala/io/gatling/jms/action/JmsRequestSend.scala

```diff
@@ -0,0 +1,144 @@
+package io.gatling.jms.action
+
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.jms.Message
+
+import scala.util.Try
+import scala.util.control.NonFatal
+
+import io.gatling.commons.stats.{ KO, OK, Status }
+import io.gatling.commons.util.TimeHelper.nowMillis
+import io.gatling.commons.validation._
+import io.gatling.core.action._
+import io.gatling.core.check.Check
+import io.gatling.core.session.{ Expression, Session, resolveOptionalExpression }
+import io.gatling.core.stats.StatsEngine
+import io.gatling.core.util.NameGen
+import io.gatling.jms.client.JmsClient
+import io.gatling.jms.protocol.JmsProtocol
+import io.gatling.jms.request._
+import io.gatling.jms._
+
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import io.gatling.core.stats.message.ResponseTimings
+
+/**
+ * Core JMS Action to handle Request-Reply semantics
+ *
+ * This handles the core "send"ing of messages. Gatling calls the execute method to trigger a send.
+ * This implementation then forwards it on to a tracking actor.
+ */
+object JmsRequestSend extends NameGen {
+
+  def apply(attributes: JmsAttributes, protocol: JmsProtocol, system: ActorSystem, statsEngine: StatsEngine, next: Action) = {
+    val actor = system.actorOf(JmsRequestSendActor.props(attributes, protocol, statsEngine, next))
+    new ExitableActorDelegatingAction(genName("jmsRequest"), statsEngine, next, actor)
+  }
+}
+
+object JmsRequestSendActor {
+
+  def props(attributes: JmsAttributes, protocol: JmsProtocol, statsEngine: StatsEngine, next: Action) =
+    Props(new JmsRequestSendActor(attributes, protocol, statsEngine, next))
+}
+
+class JmsRequestSendActor(attributes: JmsAttributes, protocol: JmsProtocol, val statsEngine: StatsEngine, val next: Action)
+    extends ValidatedActionActor {
+
+  // Create a client to refer to
+  val client = JmsClient(protocol, attributes.destination, attributes.replyDestination)
```

Comment:
The problem with the JMSClient is that the whole client is build for reply staff. So in general the JmsClient must be rewritten and there should be a JmsReplyClient as specialization.
What do you think?